### ISSUE TITLE
[Bug] After deleting the last subdeck, the deck under the last-deleted subdeck is shown as if it were a subdeck

### ISSUE THREAD

**Issue by snowtimeglass** [ankidroid/Anki-Android#10636]
Thanks to developers.

###### Reproduction Steps

1. Delete the last subdeck* on `Decks` list

   *This "the last subdeck" means something like this: "After the subdeck is deleted, there aren't any subdecks on `Decks` list. In other words, after the subdeck is deleted, there aren't any `∨` (or `>`) buttons for subdecks on `Decks` list."

    
    ![image](https://user-images.githubusercontent.com/10436072/160271540-4df491fa-28f4-4ae9-a0a8-12217ec61099.png)

###### Expected Result
 All decks line up vertically.
    ![image](https://user-images.githubusercontent.com/10436072/160271475-49d89f5e-e32a-4e3c-8168-a00169d2d124.png)

###### Actual Result

 The deck under the last-deleted subdeck doesn't get in the line. The deck is shown as if it were a subdeck.
    ![image](https://user-images.githubusercontent.com/10436072/160271721-84deaa25-831b-40e1-975e-e27b969cd574.png)

(After closing and reopening AnkiDroid, the deck gets in the line.)

https://user-images.githubusercontent.com/10436072/160271981-c1f31cca-3d20-469c-9393-f69fa4653f2c.mp4

###### Debug info
```
AnkiDroid Version = 2.16alpha53

Android Version = 9

Manufacturer = OPPO

Model = CPH1983

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 9; CPH1983 Build/PKQ1.190519.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/99.0.4844.58 Mobile Safari/537.36

ACRA UUID = 057cd16c-e7f1-4e0c-a259-8401e998e532

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true
```

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by A-Deekshith** [comment]
Can I work on this?

**Comment by david-allison** [comment]
Sounds great to me!

**Comment by Arthur-Milchior** [comment]
Thanks a lot @snowtimeglass for this very detailed report. Screen capture, reproduction, and even highlighting on the screenshot, I'm really amazed by the effort you took.

Can I please ask one last thing, that may potentially help. If you review a deck and come back to deck-picker, does it go away? If you wait for a minute, does it go away (the deck picker usually refresh every 30 seconds I believe)

**Comment by Arthur-Milchior** [comment]
Oh, actually, the first two screenshot show that all decks moves, EXCEPT deck three. If any deck has a subdeck, there is a left column for > and V. If no deck has subdeck, then this column is not present. This means that all deck names are slightly more on the right when there is at least one subdeck somewhere than when there is no subdeck.
In the case you display, it seems that for some reason, Deck 3 believes there is subdeck and keep its previous padding while all other decks are updated.
I suspect that you want to look at `onBindViewHolder` in `DeckAdapter` and see why some padding is added.

My first intuition was that the lines of the list were reused, and some properties were not reset. That often occurs when you code and expect to get fresh object, and instead the API gives you objects already used; which already created issues in the browser content if I recall correctly

**Comment by snowtimeglass** [comment]
@Arthur-Milchior thanks always, and thanks so much for the comment. I'm so glad to hear that :-)

>Can I please ask one last thing, that may potentially help. 

With pleasure.

>If you review a deck and come back to deck-picker, does it go away? If you wait for a minute, does it go away (the deck picker usually refresh every 30 seconds I believe)

As @A-Deekshith says, misalignment isn't refreshed after waiting for a minute in Study screen on my device either, but, after hiding AnkiDroid app to the background and reopening it, the deck gets in the line as in the following video. It was strange for me because I had thought that just hiding and reopening the App can't make the alignment. I can't reproduce this phenomenon yet.  

https://user-images.githubusercontent.com/10436072/162217185-755a24e4-f79c-4479-8d8f-7a25c6bed86b.mp4

>If any deck has a subdeck, there is a left column for > and V. If no deck has subdeck, then this column is not present. This means that all deck names are slightly more on the right when there is at least one subdeck somewhere than when there is no subdeck.

Yes, I think that's right.

>actually, the first two screenshot show that all decks moves, EXCEPT deck three. 

>In the case you display, it seems that for some reason, Deck 3 believes there is subdeck and keep its previous padding while all other decks are updated.

I have to say that these are not true. `Deck 3` also moves to the left as follows. It moves about a half of the length of the others, oddly. 
![image](https://user-images.githubusercontent.com/10436072/162231699-92b59f81-1c34-44ba-a6d4-44fdd978afc0.png)

https://user-images.githubusercontent.com/10436072/162227645-8f0cb178-b23c-4887-b879-223ac2597024.mp4

Anyway thanks! I'm sure that with these lively exchange of opinions, you developers will find a good solution. I would like to support you developers as much as I can.

**Comment by A-Deekshith** [comment]
<b>While I am working on this issue I noticed these things:</b>
1) Even after using Pull To Refresh misalignment is not getting fixed.
2) The misalignment is happening to the next Deck which is next to the deleted subdeck.

https://user-images.githubusercontent.com/77965954/162573472-2fe7b8ad-90d2-4ca9-acea-c7462fafb434.mp4

Any Suggestions to work on?

**Comment by Arthur-Milchior** [comment]
> Any Suggestions to work on?

I already gave one. Where you able to look at the function I mentioned, understand what it does, check whether anything was wrong here? It's hard to give you suggestion if you don't provide feedback on the first suggestion I gave. I don't see how to suggest more unless I go and try to do the work myself (and checking which method may be related is already an important part of the process)

All of those new informations are extremly interesting to understand more of what is going on, for sure, to try to read the code and see where the current behavior is implemented, but still, my first suspicion remains the same
